### PR TITLE
Added test cases newlines in string interpolation C#11 feature

### PR DESCRIPTION
**Pull request by cristian-ambrosini-sonarsource**

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/DoNotHardcodeCredentials_DefaultValues.CSharp11.cs [hunk 1]
```diff
@@ -1,16 +1,33 @@
...
+    var DBConnectionString3 = """
+        Server=localhost; Database=Test; User=SA; Password=Secret123
+        """u8; // FN
+}
+
+void NewlinesInStringInterpolation(string someInput)
+{
+    const string test1 = "test1";
+    const string test2 = "test2";
+    string noncompliant = $"Server = localhost; Database = Test; User = SA; Password ={test1 // Noncompliant
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/DoNotHardcodeCredentials_DefaultValues.CSharp11.cs:29]: Also here, I would move the comments out of the string.

**Review by costin-zaharia-sonarsource — CHANGES_REQUESTED** [review]
LGTM! just a few small comments

**Comment by gregory-paidis-sonarsource** [comment]
Found one more rule that _might_ interest us:

## DoNotLockOnSharedResource

```csharp
            var foo = 1;

            lock ($"{
                foo
                }")
            { }
            // Noncompliant@-3
```

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Review by gregory-paidis-sonarsource — APPROVED** [review]

**Pull request merged**